### PR TITLE
Use 'tox' to run tests

### PR DESCRIPTION
**Pull request by jml**
Allows for testing multiple Pythons.

- [x] make sure tests pass
- [x] update changelog
- [ ] use six

Builds on #107

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -23,56 +23,53 @@ DOCKER_IMAGE_DIRS=$(patsubst %/Dockerfile,%,$(DOCKERFILES))
...
 
-lint: $(VIRTUALENV_BIN)/flake8
-	$(VIRTUALENV_BIN)/flake8 gfdatasource/gfdatasource grafanalib
+lint: .ensure-flake8
+	$(FLAKE8) gfdatasource/gfdatasource grafanalib
 
-test: $(VIRTUALENV_BIN)/py.test
-	$(VIRTUALENV_BIN)/py.test --junitxml=$(JUNIT_XML)
+test: .ensure-tox
+	$(TOX) --skip-missing-interpreters
```
Comment (leth) [inline Makefile]: I can see why you'd want to skip when developing locally, but will they be skipped by CI?

**Comment by leth** [comment]
Regarding six: I suggest you vendor it into the project, rather than adding it as dependency.
The whole world tends to depend on six, which creates a version constraint satisfication nightmare.

**Comment by jml** [comment]
Thanks! We don't yet need six (according to our tests), so I'm skipping that for now.

File: Makefile (hunk 1 above)
Comment (jml) [reply Makefile]: Yes, probably. I'm going to risk it though.

**Pull request merged**